packages/sources/coinmetrics/src/transport/price-ws.ts

```diff
@@ -96,12 +90,16 @@ export const handleAssetMetricsMessage = (
     // Is WsAssetMetricsErrorResponse
     logger.error(message, `Error response from websocket`)
 
-    const regex = /'([^']+)'/g
+    const findBaseCurrenciesRegex = new RegExp(/'([^']+)'/g)
     if (message['error']['type'] === 'bad_parameters') {
-      const matches = [...message['error']['message'].matchAll(regex)]
+      //Bad Parameter Error Message
+      //   {error: {
+      //     type: "bad_parameter",
+      //     message: "Metric 'ReferenceRateBTC' with frequency '1s' is not supported for 'cron'."}}
+      const matches = [...message['error']['message'].matchAll(findBaseCurrenciesRegex)]
 
-      if (matches && !invalid_base_currencies.includes(matches[2][1])) {
-        invalid_base_currencies.push(matches[2][1])
+      if (matches && !invalidBaseAssets.includes(matches[2][1])) {
+        invalidBaseAssets.push(matches[2][1])
       }
```

Comment:
If we replace `matchAll` with `match`, then we don't need the second array accessor. 

We can then do the below, which is easier to read:
```
const [metric, frequency, asset] = [...message.error.message.match(findBaseCurrenciesRegex)]
```

Reply:
This actually results in the single quotes being included, so better off continuing to use `matchAll`:

```
const [metric, frequency, asset] = [...message.error.message.matchAll(findBaseCurrenciesRegex)].map(match => match[1])
```